app/actions/sessions.js

```diff
@@ -133,7 +133,7 @@ export function saveConnections(connectionsObject, tabId) {
             tabId,
             connectionsObject
         )).then(json => {
-            dispatch(mergeTabMap({[tabId]: json.connectionId}));
```

Comment:
I find this a little bit confusing. `tabMap` is supposed to be a map between the tab IDs and the saved `connectionId`s. If the connection ID wasn't saved, then it's not supposed to have an entry inside the `tabMap`.

so is this line implicitly handling the error case for the save connection? If so, I don't think that we should be mapping the `tabId` to the `tabId` - that would imply that the `tabId` has been saved OK and that the connection's ID is actually `tabId`.

if this was injected just to handle the error case, where the request might not successfully return with a `json.connectionId`, can we be more explicit about it and do something like this?
```
if (!json.error) {
   dispatch(mergeTabMap({[tabId]: json.connectionId}))
}
```